com.unity.cinemachine/Runtime/Behaviours/CinemachineFreeLook.cs

```diff
@@ -434,46 +434,67 @@ float GetYAxisClosestValue(Vector3 cameraPos, Vector3 up)
                 Vector3 flatDir = dir; flatDir.y = 0;
                 if (!flatDir.AlmostZero())
                 {
-                    float angle = Vector3.SignedAngle(flatDir, Vector3.back, Vector3.up);
+                    float angle = UnityVectorExtensions.SignedAngle(flatDir, Vector3.back, Vector3.up);
                     dir = Quaternion.AngleAxis(angle, Vector3.up) * dir;
                 }
                 dir.x = 0;
 
-                // Sample the spline in a few places, find the 2 closest, and lerp
-                int i0 = 0, i1 = 0;
-                float a0 = 0, a1 = 0;
-                const int NumSamples = 13;
-                float step = 1f / (NumSamples-1);
-                for (int i = 0; i < NumSamples; ++i)
-                {
-                    float a = Vector3.SignedAngle(
-                        dir, GetLocalPositionForCameraFromInput(i * step), Vector3.right);
-                    if (i == 0)
-                        a0 = a1 = a;
-                    else
-                    {
-                        if (Mathf.Abs(a) < Mathf.Abs(a0))
-                        {
-                            a1 = a0;
-                            i1 = i0;
-                            a0 = a;
-                            i0 = i;
-                        }
-                        else if (Mathf.Abs(a) < Mathf.Abs(a1))
-                        {
-                            a1 = a;
-                            i1 = i;
-                        }
-                    }
-                }
-                if (Mathf.Sign(a0) == Mathf.Sign(a1))
-                    return i0 * step;
-                float t = Mathf.Abs(a0) / (Mathf.Abs(a0) + Mathf.Abs(a1));
-                return Mathf.Lerp(i0 * step, i1 * step, t);
+                // We need to find the minimum of the angle of function using steepest descent
+                return SteepestDescent(0f, 1f, dir, cameraPos);
             }
             return m_YAxis.Value; // stay conservative
         }
 
+        float SteepestDescent(float min, float max, Vector3 desiredDirection, Vector3 cameraPos)
+        {
+            var rigToCameraMatrix = Matrix4x4.TRS(
+                Follow.position, 
+                GetRig(1).GetCinemachineComponent<CinemachineOrbitalTransposer>().GetReferenceOrientation(State.ReferenceUp), 
```

Comment:
Is there a better way to get the orientation?

Reply:
This way is fine

Reply:
Wait a minute - I may have been too hasty to approve this.  It's not correct, I think. The objective should be to cancel out the effect of the X axis value, and that's not what you're doing here.